### ISSUE TITLE
Group related tests into files

### ISSUE THREAD

**Issue by maclover7** [nodejs/http-parser#400]
`test.c` is over four thousand lines long and is getting a little hard to parse... any objections to moving some related tests into separate files in a `test/` directory?

@indutny @bnoordhuis

**Comment by bnoordhuis** [comment]
Personally, I'm -0.5.  Long files don't really bother me and breakups are annoying with `git blame`.  I also don't really enjoy reviewing huge diffs.

But it Fedor thinks it's a good idea, I won't block it.

**Comment by maclover7** [comment]
ping @indutny

**Comment by indutny** [comment]
+1 for this, sorry for delay. This may require writing a mini test framework, though.
